### ISSUE TITLE
Custom Options on Configurable Products not working

### ISSUE THREAD

**Issue by revlis-x** [vuestorefront/vue-storefront#2141]
## Current behavior
I have Magento Configurable Product which also has Custom Options on the parent. This works fine in Magento, but in VSF even though the custom options are displayed the attribute selection of the child products dissappear since `product.errors` is populated with `{ "custom_options": null, "custom_options_customOption_XX": null }. `

![image](https://user-images.githubusercontent.com/2040118/50822005-ebfb8c80-1330-11e9-96f8-2f8cd715c422.png)

## Expected behavior
Magento-like behaviour where one can have both working Custom Options on Configurable Products.

## Steps to reproduce the issue
1. Add custom options to Configurable Product in Magento

## Additional information
I don't know wether this is a bug/misconfiguration or this setup is generelly not supported by VSF. Please mark as Feature Request if the latter.

**Comment by pkarw** [comment]
Hi thanks for this issue. The configurableOptions are currently suported only on the top level; so they're not supported on the configurable_children level. This is a feature request in that case

**Comment by alex24409331** [comment]
Also you may try this tutor. https://www.mag-manager.com/useful-articles/tipstricks/simple-products-with-custom-options-vs-configurable-products/

**Comment by SodHub** [comment]
Any updates on this? We need to use custom options with configurable and bundle products

**Comment by andrzejewsky** [comment]
@SodHub Hello, we don't support this for configurable products yet, it works just for simple ones. Unfortunately, implementation of that is not on our roadmap now so for sure we can't deliver in upcoming releases unless someone from the community will prepare a pull request.

**Comment by simonmaass** [comment]
@andrzejewsky @pkarw sorry i am not sure i am understanding this right...

I currently have a configurable product (with 2 configurable attributes) and the configurable product also has custom options on top level. Is this supported?

If I understand @pkarw correct "they're not supported on the configurable_children level" - then it is supported on parent product level....

But if I read @andrzejewsky "we don't support this for configurable products yet, it works just for simple ones" then that is a different answer...

Could you please clarrify this - thank you! Using magento 2 btw...

**Comment by netzkollektiv** [comment]
This seems to be already implementend but commented. Not fully tested, but works at first sight:
```
$ git diff core/modules/catalog/helpers/index.ts
diff --git a/core/modules/catalog/helpers/index.ts b/core/modules/catalog/helpers/index.ts
index 57e5c37c7..e516d763b 100644
--- a/core/modules/catalog/helpers/index.ts
+++ b/core/modules/catalog/helpers/index.ts
@@ -331,9 +331,9 @@ function _prepareProductOption (product) {
       bundle_options: []
     }
   }
-  /* if (product.product_option) {
+  if (product.product_option) {
     product_option = product.product_option
-  } */
+  }
   return product_option
 }
 export function setConfigurableProductOptionsAsync (context, { product, configuration }) {
```

**Comment by simonmaass** [comment]
so i tried to look into this a bit more...

core\modules\catalog\helpers\productOptions\setProductConfigurableOptions.ts

here in line 16 this is set:
`const productOptions = getAllProductConfigurations({ configurableOptions, configuration })`
in these productOptions the custom_options are not added!

also when updating the custom_options it deletes the key 'current.product_options.extension_attributes.configurable_item_options

Unfortunately i am not 100% able to fix this problem and would need some help @gibkigonzo @pkarw  any chance this will be fixed/added to 1.12.2?

This is a standard feature in magento 2 - to have custom options on an configurable product - would be great if vsf would support this aswell! I think this feature doesnt need a lot of fixing in order to make it work...

**Comment by netzkollektiv** [comment]
There is also an issue when loading a cart including such a product from the server (client cart empty). The configurable child is not being found in productActions.ts `findProductOption` and thus (silently) not being added in the cart.
